### ISSUE TITLE
Is it Really work 

### ISSUE THREAD

**Issue by saroar** [gperdomor/sygnaler#14]
i am trying to run this app on Heroku but does success so can help to fixed issue and also try your self to push it to Heroku thanks

**Comment by gperdomor** [comment]
@saroar Hi... thanks for your interest... Sadly this project was made for play with matrix and test how works vapor in a real project to promote the adoption of the framework in our company... Notice that the notifications only works for Apple Devices... Android notifications are not supported  for now...

I will try to find some time to rewrite to Vapor 3 :D

The problem with Heroku should be the HTTP2 protocol which is not supported in Heroku, or at least was not supported a year ago... Curl with HTTP2 is required by one of the dependencies to send notifications to APNS servers

**Comment by saroar** [comment]
@gperdomor thanks for ur reply 
can we try perfect framework i know little bit

**Comment by saroar** [comment]
is vapor it ready to production real app

**Comment by gperdomor** [comment]
I never used Perfect... Only Vapor, I try to follow their evolution close and I have some package repos related to it. Yes, Vapor is production ready, just now the team release the Version 3 with a complete rewrite and a new NIO (Non Blocking IO) architecture... Why are you interested in this project? Are you working with the Matrix Protocol?

**Comment by saroar** [comment]
yes i clone RIOT app and trying to run it for one small company but my notification dont work into my ios app :( 4 days already 

can you help with notification?

**Comment by saroar** [comment]
i create certificate as this guide 
http://keyvanfatehi.com/2017/03/27/How-to-configure-Matrix-Sygnal-push-gateway-server-with-Apple/
but does not helping me

**Comment by gperdomor** [comment]
You can't use a custom RIOT with the official matrix server and get notifications... If you need notifications you need run your own server and notification gateway... This https://github.com/matrix-org/synapse for the server and for notification you can use this project or https://github.com/matrix-org/sygnal

**Comment by saroar** [comment]
https://github.com/matrix-org/sygnal yes my back-end dev using this app but we getting SSL error

**Comment by saroar** [comment]
also we have our own server too

**Comment by gperdomor** [comment]
who send json payloads to sygnal? the matrix server or your dev using a rest client? maybe you can try the last option first to see what happen

**Comment by saroar** [comment]
our dev using a rest client

**Comment by saroar** [comment]
which will be rit 

```
ru.vistamed.dev.type = apns
ru.vistamed.dev.platform = sandbox < ?
ru.vistamed.dev.certfile = /root/.sygnal/apns.pem
```
```
ru.vistamed.dev.type = apns
ru.vistamed.dev.platform = production < ?
ru.vistamed.dev.certfile = /root/.sygnal/apns.pem
